### ISSUE TITLE
is it training error or training is running ??

### ISSUE THREAD

**Issue by AnsMumtaz** [cloud-annotations/training#60]
<img width="1065" alt="Screenshot 2019-04-29 at 23 00 32" src="https://user-images.githubusercontent.com/43657814/56927433-76305200-6ad4-11e9-9a11-67c97a807bb0.png">

1) i started training and it says its running... i have waited for 40min but it still show me the same thing... i have no idea what to do ??

2) how can i ensure that my annotation are save... i mean annotation file and its address ??

**Comment by bourdakos1** [comment]
It’s probably not training it looks like it crashed can you send the entire message

**Comment by AnsMumtaz** [comment]
<img width="963" alt="Screenshot 2019-05-02 at 14 27 12" src="https://user-images.githubusercontent.com/43657814/57075220-799b2780-6ce6-11e9-87a2-4868273e09ce.png">

this is the entire message

**Comment by bourdakos1** [comment]
What version of node do you have?

**Comment by AnsMumtaz** [comment]
v10.15.3

**Comment by bourdakos1** [comment]
How many times have you tried to run? And can you send a copy of you config.yaml?

**Comment by AnsMumtaz** [comment]
i think this is probably the same issue because nothing is stored in my
config.yaml file

**Comment by AnsMumtaz** [comment]
i have tried 2,3 times

**Comment by bourdakos1** [comment]
What version of cacli do you have? Run “cacli”

**Comment by bourdakos1** [comment]
You can try running “cacli init” too to refill the config.yaml

**Comment by AnsMumtaz** [comment]
i have config my data....i have attached the config.yaml file i dont know
thats the problem...

**Comment by AnsMumtaz** [comment]
same error is coming ... i have sent the config.yaml file
and also my cacli version is 1.0.14

**Comment by bourdakos1** [comment]
If you make a localization bucket, will that train?

**Comment by AnsMumtaz** [comment]
I have already made localization bucket.

**Comment by AnsMumtaz** [comment]
i have localization bucket... and its not training...

**Comment by bourdakos1** [comment]
Oh sorry I got this confused with someone else. How many images do you have labeled?

**Comment by AnsMumtaz** [comment]
i have 4 different classes speed20, 30, 50and 60 and i have labelled about
350 images

**Comment by bourdakos1** [comment]
Did you draw rectangles?

**Comment by AnsMumtaz** [comment]
Yes

**Comment by bourdakos1** [comment]
Not sure then...

**Comment by bourdakos1** [comment]
If you send me an email with your credentials I can try and see whats going on

**Comment by Unyxos** [comment]
Hi,

I'm running into the same issue while following this guide: https://cloud-annotations.github.io/training/object-detection/cli/7.html

I'm running on Windows, node 11.14.0, config.yaml isn't empty and all the data I entered should be correct...
I have 30 labeled images (just like in the guide)

**Comment by bourdakos1** [comment]
same status code 500 UNAVAILABLE error? Do you also want to email me your credentials?

**Comment by RWOverdijk** [comment]
Exactly what @Unyxos says is happening here. Also following that guide. I have three labels "open", "closed" and "neither". 

```
(Using settings from config.yaml)
⠦ Starting training run...(node:30708) UnhandledPromiseRejectionWarning: StatusCodeError: 500 - {"trace":"38a44e27c21b3e95ff144daf8130d626","errors":[{"code":"training_submit_error","message":"UNAVAILABLE: io exception","more_info":"http://watson-ml-api.mybluemix.net/"}]}
    at new StatusCodeError (/Users/rwoverdijk/.config/yarn/global/node_modules/cloud-annotations/node_modules/request-promise-core/lib/errors.js:32:15)
    at Request.plumbing.callback (/Users/rwoverdijk/.config/yarn/global/node_modules/cloud-annotations/node_modules/request-promise-core/lib/plumbing.js:104:33)
    at Request.RP$callback [as _callback] (/Users/rwoverdijk/.config/yarn/global/node_modules/cloud-annotations/node_modules/request-promise-core/lib/plumbing.js:46:31)
    at Request.self.callback (/Users/rwoverdijk/.config/yarn/global/node_modules/request/request.js:185:22)
    at Request.emit (events.js:182:13)
    at Request.EventEmitter.emit (domain.js:442:20)
    at Request.<anonymous> (/Users/rwoverdijk/.config/yarn/global/node_modules/request/request.js:1161:10)
    at Request.emit (events.js:182:13)
    at Request.EventEmitter.emit (domain.js:442:20)
    at IncomingMessage.<anonymous> (/Users/rwoverdijk/.config/yarn/global/node_modules/request/request.js:1083:12)
(node:30708) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:30708) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
⠸ Starting training run...
```

My config was generated by the cli:

```
name: mouth
credentials:
  wml:
    instance_id: 04044cc5-e39d-****
    username: 205c4719-97df-4ecc-****
    password: 5d146b7a-9c46-49fd-92e8-****
    url: 'https://eu-de.ml.cloud.ibm.com'
  cos:
    access_key_id: 2632a254063244****
    secret_access_key: 1c145c285d91d****
    region: us-geo
buckets:
  training: mouth
  output: mouth-models
trainingParams: {}
```

**Comment by RWOverdijk** [comment]
Not sure if this is relevant... But I tried using the watson dashboard itself and it also breaks there:

```html
Error encountered while training.

Error in Watson Visual Recognition service: <html>
 <head><title>413 Request Entity Too Large</title></head>
 <body bgcolor="white">
 <center><h1>413 Request Entity Too Large</h1></center>
 <hr><center>nginx</center>
 </body>
 </html>
 <!-- a padding to disable MSIE and Chrome friendly error page -->
 <!-- a padding to disable MSIE and Chrome friendly error page -->
 <!-- a padding to disable MSIE and Chrome friendly error page -->
 <!-- a padding to disable MSIE and Chrome friendly error page -->
 <!-- a padding to disable MSIE and Chrome friendly error page -->
 <!-- a padding to disable MSIE and Chrome friendly error page -->
```

![image](https://user-images.githubusercontent.com/781745/57364967-7a1f3c80-7184-11e9-8db4-e4b1d1e31b76.png)

**Comment by bourdakos1** [comment]
@deerdear I'll try to reproduce the issue, did you use an output bucket?

**Comment by bourdakos1** [comment]
@deerdear I'll try to reproduce, did you use an output bucket?

**Comment by bourdakos1** [comment]
@deerdear I'm trying to reproduce the issue, did you use an output bucket? 

p.s. if you get like 30 of these messages it's because github says they aren't sending

**Comment by tristanisolde** [comment]
> And I can't download the model, either since it didn't work. I wasn't told upon failure though but the only message I got after the training finished was:
> 
> ```
> Would you like to monitor progress? (yes)
> 
> success Training in progress.
> success Training complete.
> success Model files saved to bucket.
> ✨ Done.
> ```
> 
> Maybe adding an error message if training exited with an error code would be helpful?

Same here, will run again and share the logs

**Comment by bourdakos1** [comment]
I'm having the same issue, I'll look deeper into it today

**Comment by bourdakos1** [comment]
@deerdear I’m getting the same issue, I look deeper into what’s going on

**Comment by bourdakos1** [comment]
I'm getting the same issue, I'll try to fix this

**Comment by bourdakos1** [comment]
The issue is fixed, but I need to write some tests before I release an update :)

**Comment by fjoalland** [comment]
Thanks to your amazing job 👍 

Same error than @deerdear when I display the log.

Result of **calci list**
![image](https://user-images.githubusercontent.com/6407511/57535056-bb108000-7341-11e9-90d8-7199b4925b42.png)

Error when I display the log
![image](https://user-images.githubusercontent.com/6407511/57535183-fe6aee80-7341-11e9-97df-5128073d2040.png)

Tell us when you deploy the fix.

**Comment by bourdakos1** [comment]
@fjoalland @deerdear @tristanisolde This should be fixed with `v1.1.0` I also display error messages during the training process if any occur, instead of always displaying `success`

**Comment by deerdear** [comment]
Awesome!!!
